core/modules/layout/plugins/access/date_time_layout_access.inc

```diff
@@ -0,0 +1,75 @@
+<?php
+/**
+ * @file
+ * Plugin to provide access control/visibility based on date and time.
+ */
+class DateTimeLayoutAccess extends LayoutAccessNegatable {
+
+  /**
+   * {@inheritdoc}
+   */
+  function summary() {
+    $from = format_date(strtotime($this->settings['start_time']),'medium');
+    $to = format_date(strtotime($this->settings['end_time']),'medium');
+    $range = t('from ') . $from . t(' to ') . $to; 
+    if ($this->settings['negate']) {
+      return t('Date is NOT: @range', array('@range'  => $range));
+    }
+    else {
+      return t('Date is: @range', array('@range'  => $range));
+    }
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  function checkAccess() {
+    $timezone = '';
+    if ($this->settings['timezone'] == 'user') {
+      $timezone = ' ' . backdrop_get_user_timezone();
+    }
+    $from = strtotime($this->settings['start_time'] . $timezone);
+    $to = strtotime($this->settings['end_time'] . $timezone);
+    $now = time();
+    $in_range = (($from < $now) && ($to > $now));
+
+    return $this->settings['negate'] ? !$in_range : $in_range;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  function form(&$form, &$form_state) {
+    parent::form($form, $form_state);
+
+    $now = format_date(time(),'medium');
+    $format = 'Y-m-d H:i';
+     
+    $form['start_time'] = array(
+      '#type' => 'date_popup',
+      '#date_format' => $format,
+      '#title' => t('Start date'),
+      '#date_label_position' => 'invisible',
+      '#date_year_range' => '-0:+3',
+      '#default_value' => isset($this->settings['start_time']) ? $this->settings['start_time'] : '',
+    );
+    $form['end_time'] = array(
+      '#type' => 'date_popup',
+      '#date_format' => $format,
+      '#title' => t('End date'),
+      '#date_label_position' => 'invisible',
+      '#date_year_range' => '-0:+3',
+      '#default_value' => isset($this->settings['end_time']) ? $this->settings['end_time'] : '',
+    );
```

Comment:
Neither start nor end times are required here. What happens if these values are left empty?

I actually think it might be a good idea to support an optional end date. e.g. "Do not show if date is after X" or "Show if date is after X". But I can't figure how an optional start date would work if an end date were specified.